python/nav/mibs/itw_mib.py

```diff
@@ -32,947 +31,435 @@
-                result.extend(method(sensors))
+        for table, sensor_groups in self.TABLES.items():
+            self._logger.debug('get_all_sensors: table = %s', table)
+            sensors = yield self.retrieve_table(
+                                        table).addCallback(reduce_index)
+            self._logger.debug('get_all_sensors: %s = %s', table, sensors)
+            for sensor_group in sensor_groups:
+                result.extend(self._handle_sensor_group(sensor_group, sensors))
 
         defer.returnValue(result)
 
-    @classmethod
-    def translate_counter_to_table(cls, counter_name):
-        """Translates the name of a count object under sensorCounts into its
-        corresponding sensor table object name.
-
-        If unable to translate the counter name into a table name, None is
-        returned.
-
-        """
-        counter_bases = [counter_name.replace('Count', '').lower(),
-                         counter_name.replace('SensorCount', '').lower()]
-        suffixes = ['table', 'sensortable', 'monitortable']
-        alternatives = [base + suffix
-                        for base in counter_bases for suffix in suffixes]
 
-        for alt in alternatives:
-            if alt in cls.lowercase_nodes:
-                return cls.lowercase_nodes[alt]
+UNITS = {
+    '%': {'u_o_m': Sensor.UNIT_PERCENT},
+    '0.1 Amps': {
+        'u_o_m': Sensor.UNIT_AMPERES,
+        'precision': 1,
+    },
+    '0.1 Amps (rms)': {
+        'u_o_m': Sensor.UNIT_AMPERES,
+        'precision': 1,
+    },
+    '0.1 Degrees': {
+        'u_o_m': Sensor.UNIT_CELSIUS,
+        'precision': 1,
+    },
+    'Degrees Celsius': {
+        'u_o_m': Sensor.UNIT_CELSIUS,
+    },
+    'kWh': {
+        'u_o_m': Sensor.UNIT_WATTHOURS,
+        'scale': Sensor.SCALE_KILO,
+    },
+    'millivolts': {
+        'u_o_m': Sensor.UNIT_VOLTS_DC,
+        'scale': Sensor.SCALE_MILLI,
+    },
+    'Volt-Amps': {
+        'u_o_m': Sensor.UNIT_VOLTAMPERES,
+    },
+    'Volts': {
+        'u_o_m': Sensor.UNIT_VOLTS_DC,
+    },
+    'Volts (rms)': {
+        'u_o_m': Sensor.UNIT_VOLTS_AC,
+    },
+    'Watts': {
+        'u_o_m': Sensor.UNIT_WATTS,
+    },
+}
+
+
+def getRange(mib, node):
+    res = {}
+    range_ = mib['nodes'][node].get('syntax', {}).get('type', {}).get('range')
+    if not range_:
+        return res
+    if 'min' not in range_ or 'max' not in range_:
+        return res
+    res['minimum'] = range_['min']
+    res['maximum'] = range_['max']
+    return res
+
+
+def convertUnits(mib, node):
+    unit = mib['nodes'][node].get("units")
+    if unit:
+        if unit in UNITS:
+            res = UNITS[unit].copy()
+            if res['u_o_m'] == Sensor.UNIT_PERCENT and "Humid" in node:
+                res['u_o_m'] = Sensor.UNIT_PERCENT_RELATIVE_HUMIDITY
+            return res
+    if 'syntax' not in mib['nodes'][node]:
+        return {'u_o_m': Sensor.UNIT_TRUTHVALUE}.copy()
+    return {'u_o_m': Sensor.UNIT_UNKNOWN}.copy()
```

Comment:
Not entirely sure what the value of copying a newly-constructed dict in these two lines is?

Reply:
Just to parallel the copy of the dict from a constant a few lines up, where the copy is necessary, as the dict does get changed later on. These two, however, does not strictly need to be copied